modules/src/ics18_relayer/utils.rs

```diff
@@ -77,10 +78,20 @@ mod tests {
         let client_on_b_for_a = ClientId::from_str("ibczeroclient").unwrap();
 
         // Create two mock contexts, one for each chain.
-        let mut ctx_a = MockContext::new(5, chain_a_start_height)
-            .with_client(&client_on_a_for_b, client_on_a_for_b_height);
-        let mut ctx_b = MockContext::new(5, chain_b_start_height)
-            .with_client(&client_on_b_for_a, client_on_b_for_a_height);
+        let mut ctx_a = MockContext::new(
+            ChainId::new("mockgaia", 1).unwrap(),
+            HostType::Mock,
+            5,
+            chain_a_start_height,
+        )
+        .with_client(&client_on_a_for_b, client_on_a_for_b_height);
+        let mut ctx_b = MockContext::new(
+            ChainId::new("mockgaia", 1).unwrap(),
+            HostType::Mock,
```

Comment:
```suggestion
            HostType::SynteticTendermint,
```
It would be good to create different types of contexts in the test. I think there are a few places where the code still needs to be updated, e.g. `with_client_parametrized()` or `store_client_state()`.

Reply:
Modified context `ctx_b` to rely on a `HostType::SynteticTendermint` chain and updated the relevant parts.

Note that some verification functions (like [this one](https://github.com/informalsystems/ibc-rs/blob/d40a80eda432e602d51124b28abc1c262d51fd4b/modules/src/ics07_tendermint/client_def.rs#L20)) are incomplete, but will add more checks in there as we go along and write new tests.